converters/toZigbee.js

```diff
@@ -841,234 +842,93 @@ const converters = {
     light_color: {
         key: ['color'],
         convertSet: async (entity, key, value, meta) => {
-            // Check if we need to convert from RGB to XY and which cmd to use
             let command;
+            const newColor = libColor.extractColorFromValue(value);
             const newState = {};
 
+            const zclData = {transtime: utils.getTransition(entity, key, meta).time};
 
-            if (value.hasOwnProperty('r') && value.hasOwnProperty('g') && value.hasOwnProperty('b')) {
-                const xy = utils.rgbToXY(value.r, value.g, value.b);
-                value.x = xy.x;
-                value.y = xy.y;
-            } else if (value.hasOwnProperty('rgb')) {
-                const rgb = value.rgb.split(',').map((i) => parseInt(i));
-                const xy = utils.rgbToXY(rgb[0], rgb[1], rgb[2]);
-                value.x = xy.x;
-                value.y = xy.y;
-            } else if (value.hasOwnProperty('hex') || (typeof value === 'string' && value.startsWith('#'))) {
-                const xy = utils.hexToXY(typeof value === 'string' && value.startsWith('#') ? value : value.hex);
-                value = {x: xy.x, y: xy.y};
-            } else if (value.hasOwnProperty('h') && value.hasOwnProperty('s') && value.hasOwnProperty('l')) {
-                newState.color = {h: value.h, s: value.s, l: value.l};
-                const hsv = utils.gammaCorrectHSV(...Object.values(
-                    utils.hslToHSV(utils.correctHue(value.h, meta), value.s, value.l)));
-                value.saturation = utils.mapNumberRange(hsv.s, 0, 100, 0, 254);
-                value.brightness = utils.mapNumberRange(hsv.v, 0, 100, 0, 254);
-                newState.brightness = value.brightness;
-
-                if (utils.getMetaValue(entity, meta.mapped, 'enhancedHue', 'allEqual', true)) {
-                    value.hue = utils.mapNumberRange(hsv.h % 360, 0, 360, 0, 65535);
-                    command = 'enhancedMoveToHueAndSaturationAndBrightness';
-                } else {
-                    value.hue = utils.mapNumberRange(hsv.h, 0, 360, 0, 254);
-                    command = 'moveToHueAndSaturationAndBrightness';
+            if (newColor.rgb !== null || newColor.xy !== null) {
+                if (meta.state.hasOwnProperty('color')) {
+                    delete meta.state.color.hue;
```

Comment:
We should probably make this configurable

Reply:
For color_sync I used a device option but ran into some issues with groups... so on the application level might be better IMHO.

I'd expect you generally want all your bulbs to act the same.

Reply:
> I'd expect you generally want all your bulbs to act the same.

I agree, and that's why I'd opt for application level config.

Reply:
I think this should be addressed in a different pr (the color_sync thing where @sjorge is working on)

Reply:
> I believe this was not done before so please remove it.

done